jest.setup.js

```diff
@@ -1,7 +1,62 @@
 // Mock Web Media APIs
 
-global.MediaStream = jest.fn();
-global.MediaStream.prototype = {
-  getTracks: jest.fn(() => []),
-  removeTrack: jest.fn(),
-};
+global.MediaStream = jest.fn((tracksOrStream = []) => {
+  let tracks;
+
+  if (tracksOrStream instanceof global.MediaStream) {
+    tracks = [...tracksOrStream.getTracks()];
+  } else {
+    tracks = tracksOrStream;
+  }
+
+  for (const track of tracks) {
+    if (!('enabled' in track)) {
+      track.enabled = true;
+    }
+
+    if (!('stop' in track)) {
+      Object.defineProperty(track, 'stop', {
+        value: jest.fn(),
+        enumerable: false,
+      });
+    }
+  }
+
+  return {
+    getTracks: jest.fn(() => tracks),
+    getAudioTracks: jest.fn(() => tracks.filter((track) => track.kind === 'audio')),
+    getVideoTracks: jest.fn(() => tracks.filter((track) => track.kind === 'video')),
+    removeTrack: jest.fn((toRemove) => {
+      tracks.splice(0, tracks.length, ...tracks.filter((track) => !Object.is(track, toRemove)));
```

Comment:
I think it's fine to use the new array without splice. Also, you can use `track !== toRemove`